yarn-project/simulator/src/avm/journal/journal.ts

```diff
@@ -156,7 +156,16 @@ export class AvmPersistableStateManager {
   public async checkL1ToL2MessageExists(msgHash: Fr, msgLeafIndex: Fr): Promise<boolean> {
     let exists = false;
     try {
-      const gotMessage = await this.hostStorage.commitmentsDb.getL1ToL2MembershipWitness(msgHash);
+      // The following 2 values are used to compute a message nullifier. Given that here we do not care about getting
+      // non-nullified messages we can just pass in random values and the nullifier check will effectively be ignored
+      // (no nullifier will be found).
+      const ignoredContractAddress = AztecAddress.random();
+      const ignoredSecret = Fr.random();
+      const gotMessage = await this.hostStorage.commitmentsDb.getL1ToL2MembershipWitness(
```

Comment:
Yes, it's ugly but changing the function signature is impractical because it would require changing like 6 classes implementing the `CommitmentsDB` interface.